app/routes/api/users.js

```diff
@@ -104,6 +104,20 @@ router.post('/:username', function(req, res, next){
         if(req.body.email){
           user.email = req.body.email;
         }
+        if(req.body.newpass && req.body.newpass == req.body.confpass){
+          user.setPassword(req.body.newpass, function(error){
+            if(!error){
+              user.save(function(error){
+                if(error){
+                  console.log(error)
```

Comment:
Please do res.send(err) instead of console.log, so that we can catch errors in the angular app instead if needed